dlt/common/normalizers/json/relational.py

```diff
@@ -410,12 +420,18 @@ def _get_validity_column_names(schema: Schema, table_name: str) -> List[Optional
 
     @staticmethod
     @lru_cache(maxsize=None)
-    def _dlt_id_is_row_hash(schema: Schema, table_name: str) -> bool:
-        return (
-            schema.get_table(table_name)["columns"]  # type: ignore[return-value]
-            .get("_dlt_id", {})
-            .get("x-row-version", False)
-        )
+    def _get_row_id_type(schema: Schema, table_name: str) -> TRowIdType:
```

Comment:
lru cache will hash schema by id (instance reference)? right?

Reply:
Right, because `Schema` is a user-defined class. 

_"Objects which are instances of user-defined classes are hashable by default. They all compare unequal (except with themselves), and their hash value is derived from their [id()](https://docs.python.org/3/library/functions.html#id)."_

https://docs.python.org/3/glossary.html#term-hashable